### PR TITLE
Update all dependencies. Less babel'iness. Standard compliant

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

**Comment by diasdavid** [comment]
As I get things prepared to try using webui with js-ipfs, I decided to go through the repo and do a clean up that:

- Update all the dependencies
- Standard all the packages
- Structure it in our standard `src` for src

As a byproduct of that:

- No more `static` to define things like propTypes

File: test/views/object/object-link.spec.js [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
 import mh from 'multihashes'
 
-import {parse} from '../../../app/scripts/utils/path'
-import ObjectLink from '../../../app/scripts/views/object/object-link'
+import {parse} from '../../../src/app/js/utils/path'
+import ObjectLink from '../../../src/app/js/views/object/object-link'
 
 describe('ObjectLink', () => {
-  it('renders the given link', () => {
+  it.skip('renders the given link', () => {
```
Comment (diasdavid) [inline test/views/object/object-link.spec.js]: Still investigating why `el.find` is not picking up the Link elements anymore

File: test/views/object/object-link.spec.js [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
+    }} />)
+
+    // TODO fails because it doesn't find the Link property
     expect(el.find('Link')).to.have.length(2)
     expect(el.find('Link').at(0)).to.have.prop('to', 'objects/\\ipfs\\hello\\world\\hi')
     expect(el.find('Link').at(0).children()).to.have.text('hi')
   })
 
-  it('renders links without a name', () => {
+  it.skip('renders links without a name', () => {
```
Comment (diasdavid) [inline test/views/object/object-link.spec.js]: (same as above) Still investigating why `el.find` is not picking up the Link elements anymore

**Comment by diasdavid** [comment]
Other than [1](https://github.com/ipfs/webui/pull/595#discussion_r123903933) and [2](https://github.com/ipfs/webui/pull/595#discussion_r123903939) all tests pass. However, when running in dev mode I get:

image

Still investigating

File: versions/QmctngrQAt9fjpQUZr7Bx3BsXUcif52eZGTizWhvcShsjz/static/js/Detector.js [hunk 3]
```diff
@@ -5,56 +5,50 @@
 
 Detector = {
 
-  canvas : !! window.CanvasRenderingContext2D,
```
Comment (dignifiedquire) [inline versions/QmctngrQAt9fjpQUZr7Bx3BsXUcif52eZGTizWhvcShsjz/static/js/Detector.js]: The versions should be changed after release, `versions` should be included from linting etc

Comment (diasdavid) [inline versions/QmctngrQAt9fjpQUZr7Bx3BsXUcif52eZGTizWhvcShsjz/static/js/Detector.js]: reverted changes in the versions folder to preserve the hashes. Thanks for catching this!

**Comment by diasdavid** [comment]
image

Working fine again. Had to downgrade react-router

Important note: `react-router-boostrap 0.24 imports react-router 4 instead of 3 compared to react-router-boostrap 0.23.`

File: test/views/object/object-link.spec.js (hunk 1 above)
Comment (diasdavid) [inline test/views/object/object-link.spec.js]: nvm, fixed with the downgrade of react router

File: test/views/object/object-link.spec.js (hunk 2 above)
Comment (diasdavid) [inline test/views/object/object-link.spec.js]: nvm, fixed with the downgrade of react router

File: src/app/js/views/table.js [hunk 4]
```diff
@@ -24,3 +17,11 @@ default class TableView extends Component {
     )
   }
 }
+
+Table.displayName = 'Table'
```
Comment (dignifiedquire) [inline src/app/js/views/table.js]: these seem like a step back, why the change?

File: src/app/js/views/connection.js [hunk 5]
```diff
@@ -2,23 +2,15 @@ import React, {Component} from 'react'
...
+class Connection extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      open: false
+    }
+  }
 
-  _handleClick = () => {
+  _handleClick () {
```
Comment (dignifiedquire) [inline src/app/js/views/connection.js]: No! this will break things, we need the binding please keep this syntax

File: src/app/js/views/config.js [hunk 6]
```diff
@@ -3,23 +3,19 @@ import PropTypes from 'prop-types'
...
+      body: JSON.stringify(this.props.config, null, 2),
+      error: null,
+      saving: false,
+      saved: false,
+      height: 0
+    }
+  }
 
-  updateHeight = () => {
+  updateHeight () {
```
Comment (dignifiedquire) [inline src/app/js/views/config.js]: same here, you can't just change this syntax, it means that `this` will not be bound anymore

File: src/app/js/views/table.js (hunk 4 above)
Comment (diasdavid) [inline src/app/js/views/table.js]: If I understand it correctly, the only reason why the previous version was working (https://github.com/ipfs/webui/pull/595#discussion_r123906426 and https://github.com/ipfs/webui/pull/595#discussion_r123906457) is thanks to some transpilation step as JS doesn't naturally support it

image

File: src/app/js/views/connection.js (hunk 5 above)
Comment (diasdavid) [inline src/app/js/views/connection.js]: right, almost missed these react'ism.

However, sintaxwise, from what I can see in other examples, it is preferable to bind things in the constructor. Wasn't this working only due thanks to some magic transpilation step in the same way that https://github.com/ipfs/webui/pull/595#discussion_r123908171 was?

Comment (diasdavid) [inline src/app/js/views/connection.js]: Ok, confirmed. I'll bring back this ES7 feature and added standard to use babel-eslint so that it doesn't warn on it.

File: src/app/js/components/files/breadcrumbs.js [hunk 7]
```diff
@@ -4,15 +4,9 @@ import {chain, isEmpty, compact} from 'lodash-es'
...
 
 class Breadcrumb extends Component {
-  static propTypes = {
-    path: PropTypes.string.isRequired,
-    onClick: PropTypes.func.isRequired,
-    text: PropTypes.string.isRequired
-  };
-
-  _onClick = (event) => {
+  _onClick (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/breadcrumbs.js]: ⏮

File: src/app/js/components/files/tree.js [hunk 8]
```diff
@@ -38,54 +38,22 @@ const fileTarget = {
...
-    root: '/',
-    files: [],
-    selectedFiles: [],
-    onRowClick () {},
-    onRowContextMenu () {},
-    onRowDoubleClick () {}
-  };
-
-  _onTmpDirChange = ({target}) => {
+  _onTmpDirChange ({target}) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree.js]: ⏮

File: src/app/js/components/files/tree.js [hunk 9]
```diff
@@ -38,54 +38,22 @@ const fileTarget = {
...
     this.props.onTmpDirChange(target.value)
-  };
+  }
 
-  _onDirCreateBlur = (event) => {
+  _onDirCreateBlur (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree.js]: ⏮

File: src/app/js/components/files/tree.js [hunk 10]
```diff
@@ -38,54 +38,22 @@ const fileTarget = {
...
     if (isEmpty(event.target.value)) {
       this.props.onCancelCreateDir()
     } else {
       this.props.onCreateDir()
     }
-  };
+  }
 
-  _isSelected = (file) => {
+  _isSelected (file) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree.js]: ⏮

File: src/app/js/components/files/tree/row-input.js [hunk 11]
```diff
@@ -3,24 +3,8 @@ import ReactDOM from 'react-dom'
...
-    value: '',
-    onChange () {},
-    onKeyEnter () {},
-    onKeyEsc () {},
-    onBlur () {}
-  };
-
-  _onKeyUp = (event) => {
+class CreateDirInput extends Component {
+  _onKeyUp (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree/row-input.js]: ⏮

File: src/app/js/components/files/tree/row.js [hunk 12]
```diff
@@ -11,33 +11,20 @@ function renderType (type) {
...
-    // onRemoveDir: PropTypes.func.isRequired,
-    selected: PropTypes.bool
-  };
-
-  static defaultProps = {
-    selected: false
-  };
-
-  _onClick = (event) => {
+  _onClick (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree/row.js]: ⏮

File: src/app/js/components/files/tree/row.js [hunk 13]
```diff
@@ -11,33 +11,20 @@ function renderType (type) {
...
     event.preventDefault()
     this.props.onClick(this.props.file, event.shiftKey)
-  };
+  }
 
-  _onContextMenu = (event) => {
+  _onContextMenu (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree/row.js]: ⏮

File: src/app/js/components/files/tree/row.js [hunk 14]
```diff
@@ -11,33 +11,20 @@ function renderType (type) {
...
     event.preventDefault()
     this.props.onContextMenu(this.props.file, event.shiftKey)
-  };
+  }
 
-  _onDoubleClick = (event) => {
+  _onDoubleClick (event) {
```
Comment (diasdavid) [inline src/app/js/components/files/tree/row.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 15]
```diff
@@ -12,30 +12,7 @@ import ActionBar from './../components/files/action-bar'
...
-    rmTmpDir: PropTypes.func.isRequired,
-    select: PropTypes.func.isRequired,
-    deselect: PropTypes.func.isRequired,
-    deselectAll: PropTypes.func.isRequired,
-    createFiles: PropTypes.func.isRequired,
-    push: PropTypes.func.isRequired
-  };
-
-  _onRowClick = (file, shiftKey) => {
+  _onRowClick (file, shiftKey) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 16]
```diff
@@ -54,9 +31,9 @@ class FilesExplorer extends Component {
       deselectAll()
       select(filePath)
     }
-  };
+  }
 
-  _onRowContextMenu = (file, shiftKey) => {
+  _onRowContextMenu (file, shiftKey) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 17]
```diff
@@ -74,9 +51,9 @@ class FilesExplorer extends Component {
       }
       select(filePath)
     }
-  };
+  }
 
-  _onRowDoubleClick = (file, shiftKey) => {
+  _onRowDoubleClick (file, shiftKey) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 18]
```diff
@@ -90,21 +67,21 @@ class FilesExplorer extends Component {
         }
       })
     }
-  };
+  }
 
-  _onCreateDir = (event) => {
+  _onCreateDir (event) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

>

File: src/app/js/containers/files-explorer.js [hunk 19]
```diff
@@ -90,21 +67,21 @@ class FilesExplorer extends Component {
...
     this.props.createTmpDir(this.props.root)
   };
 
-  _onCreateFiles = (files) => {
+  _onCreateFiles (files) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 20]
```diff
@@ -90,21 +67,21 @@ class FilesExplorer extends Component {
...
     this.props.createFiles(this.props.root, files)
-  };
+  }
 
-  _onCancelCreateDir = (event) => {
+  _onCancelCreateDir (event) {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-explorer.js [hunk 21]
```diff
@@ -90,21 +67,21 @@ class FilesExplorer extends Component {
...
     this.props.rmTmpDir()
-  };
+  }
 
-  _onRemoveDir = () => {
+  _onRemoveDir () {
```
Comment (diasdavid) [inline src/app/js/containers/files-explorer.js]: ⏮

File: src/app/js/containers/files-preview.js [hunk 22]
```diff
@@ -55,11 +41,11 @@ class FilesPreview extends Component {
     this.props.leave()
   }
 
-  _onClose = (event) => {
+  _onClose (event) {
```
Comment (diasdavid) [inline src/app/js/containers/files-preview.js]: ⏮

File: src/app/js/pages/logs.js [hunk 23]
```diff
@@ -1,31 +1,24 @@
...
+    this.state = {
+      log: [],
+      tailing: true,
+      nonce: 0,
+      request: null,
+      stream: null
+    }
+  }
+
+  _onLogData (chunk) {
```
Comment (diasdavid) [inline src/app/js/pages/logs.js]: ⏮

File: src/app/js/pages/objects.js [hunk 24]
```diff
@@ -7,17 +7,13 @@ import {Row, Col, Button} from 'react-bootstrap'
...
-    ipfs: PropTypes.object
-  };
+  constructor (props) {
+    super(props)
+    const {params, ipfs} = props
+    this.state = this._getStateFromRoute(params, ipfs)
+  }
 
-  _getStateFromRoute = (params, ipfs) => {
+  _getStateFromRoute (params, ipfs) {
```
Comment (diasdavid) [inline src/app/js/pages/objects.js]: ⏮

File: src/app/js/pages/objects.js [hunk 25]
```diff
@@ -27,51 +23,44 @@ class Objects extends React.Component {
     }
     // reset between reloads
     return state
-  };
+  }
 
-  _updateState = () => {
+  _updateState () {
```
Comment (diasdavid) [inline src/app/js/pages/objects.js]: ⏮

File: src/app/js/pages/objects.js [hunk 26]
```diff
@@ -27,51 +23,44 @@ class Objects extends React.Component {
...
     const {params, ipfs} = this.props
     this.setState(this._getStateFromRoute(params, ipfs))
   };
 
-  _getObject = (ipfs, path) => {
+  _getObject (ipfs, path) {
```
Comment (diasdavid) [inline src/app/js/pages/objects.js]: ⏮

File: src/app/js/pages/objects.js [hunk 27]
```diff
@@ -27,51 +23,44 @@ class Objects extends React.Component {
...
-          })
+          this.setState({ permalink: permalink })
         })
       }
     }).catch((error) => this.setState({error}))
   };
 
-  _update = (event) => {
-    if (event.which && event.which !== 13) return
+  _update (event) {
```
Comment (diasdavid) [inline src/app/js/pages/objects.js]: ⏮

File: src/app/js/views/config.js (hunk 6 above)
Comment (diasdavid) [inline src/app/js/views/config.js]: ⏮

File: src/app/js/views/connection.js (hunk 5 above)
Comment (diasdavid) [inline src/app/js/views/connection.js]: ⏮

File: src/app/js/views/dhtgraph.js [hunk 28]
```diff
@@ -7,18 +7,15 @@ const radius = diameter / 2
...
+class DHTGraph extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      initialized: false
+    }
+  }
 
-  update = () => {
+  update () {
```
Comment (diasdavid) [inline src/app/js/views/dhtgraph.js]: ⏮

**Comment by diasdavid** [comment]
With babel-eslint as parser: 

```
standard: Use JavaScript Standard Style (https://standardjs.com)
  /Users/koruza/code/webui/src/app/js/components/files/breadcrumbs.js:7:3: '_onClick' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree.js:41:3: '_onTmpDirChange' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree.js:45:3: '_onDirCreateBlur' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree.js:53:3: '_isSelected' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree/row-input.js:7:3: '_onKeyUp' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree/row.js:16:3: '_onClick' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree/row.js:21:3: '_onContextMenu' is not defined.
  /Users/koruza/code/webui/src/app/js/components/files/tree/row.js:26:3: '_onDoubleClick' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:15:3: '_onRowClick' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:36:3: '_onRowContextMenu' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:56:3: '_onRowDoubleClick' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:72:3: '_onCreateDir' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:76:3: '_onCreateFiles' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:80:3: '_onCancelCreateDir' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:84:3: '_onRemoveDir' is not defined.
  /Users/koruza/code/webui/src/app/js/containers/files-preview.js:32:3: '_onClose' is not defined.
  /Users/koruza/code/webui/src/app/js/pages/logs.js:21:3: '_onLogData' is not defined.
  /Users/koruza/code/webui/src/app/js/pages/objects.js:16:3: '_getStateFromRoute' is not defined.
  /Users/koruza/code/webui/src/app/js/pages/objects.js:28:3: '_updateState' is not defined.
  /Users/koruza/code/webui/src/app/js/pages/objects.js:33:3: '_getObject' is not defined.
  /Users/koruza/code/webui/src/app/js/pages/objects.js:48:3: '_update' is not defined.
  /Users/koruza/code/webui/src/app/js/views/config.js:18:3: 'updateHeight' is not defined.
  /Users/koruza/code/webui/src/app/js/views/config.js:23:3: 'componentDidMount' is not defined.
  /Users/koruza/code/webui/src/app/js/views/connection.js:13:3: '_handleClick' is not defined.
  /Users/koruza/code/webui/src/app/js/views/dhtgraph.js:18:3: 'update' is not defined.
```

Without babel-eslint as parser: 
```
standard: Use JavaScript Standard Style (https://standardjs.com)
  /Users/koruza/code/webui/src/app/js/components/files/breadcrumbs.js:7:12: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/components/files/tree.js:41:19: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/components/files/tree/row-input.js:7:12: Parsing error: Unexpected token=
  /Users/koruza/code/webui/src/app/js/components/files/tree/row.js:16:12: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/containers/files-explorer.js:15:15: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/containers/files-preview.js:32:12: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/pages/logs.js:21:14: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/pages/objects.js:16:22: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/views/config.js:18:16: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/views/connection.js:13:16: Parsing error: Unexpected token =
  /Users/koruza/code/webui/src/app/js/views/dhtgraph.js:18:10: Parsing error: Unexpected token =
```

**Comment by diasdavid** [comment]
Update here: I've been trying to debug the lint issue for the last 40 minutes. @dignifiedquire also double checked. 

Status: Can't replicate locally (could before, but after a fresh npm install, things became fine)

**Comment by VictorBjelkholm** [comment]
Managed to reproduce this locally so gave it a shot and things are passing now. 

https://github.com/ipfs/webui/pull/595/commits/1203a691ed235e7b4442c1c24238cffc17d100c8 introduces usage of eslint directly instead of via standard, extending the `standard` and `standard-react` packages.

https://github.com/ipfs/webui/pull/595/commits/b27133881e62e4c1da5bd6f75243e0c1ce19f543 fixes some minor linting issues that were found after applying previous commit.

**Comment by VictorBjelkholm** [comment]
Ah, also, found one linting issue I was not able to fix

```
/home/ubuntu/webui/src/app/js/components/files/tree.js
  123:18  warning  'onCreateFiles' PropType is defined but prop is never used  react/no-unused-prop-types
```

Seems by passing around the props as a object directly into some third-party component confuses eslint and therefore doesn't realize that we actually use that function. Anyways, it was a error before, but unused-prop-types is not super severe, so downgraded it to warn instead of error. If you know a fix, we can turn it to error again.

**Comment by diasdavid** [comment]
thank you @VictorBjelkholm ! :)
